pkg/deploy/controller/generictrigger/controller.go

```diff
@@ -36,183 +29,19 @@ type DeploymentTriggerController struct {
 	codec runtime.Codec
 }
 
-// fatalError is an error which can't be retried.
-type fatalError string
-
-func (e fatalError) Error() string {
-	return fmt.Sprintf("fatal error handling configuration: %s", string(e))
-}
-
 // Handle processes deployment triggers for a deployment config.
 func (c *DeploymentTriggerController) Handle(config *deployapi.DeploymentConfig) error {
 	if len(config.Spec.Triggers) == 0 || config.Spec.Paused {
```

Comment:
If you got an update for a DC, you only want to enqueue the work the latest spec has an config change trigger active and if the config has changed.